rest/model/patch.go

```diff
@@ -97,8 +99,47 @@ func (p *APIParameter) ToService() patch.Parameter {
 	return res
 }
 
-// BuildFromService converts from service level structs to an APIPatch
-func (apiPatch *APIPatch) BuildFromService(p patch.Patch) error {
+type APIPatchArgs struct {
+	IncludeProjectIdentifier   bool
+	IncludeCommitQueuePosition bool
+	IncludeChildPatches        bool
+}
+
+// BuildFromService converts from service level structs to an APIPatch.
+// If args are set, includes identifier, commit queue position, and/or child patches from the DB, if applicable.
+func (apiPatch *APIPatch) BuildFromService(p patch.Patch, args *APIPatchArgs) error {
+	projectIdentifier := p.Project
+	if args != nil {
+		if args.IncludeProjectIdentifier && p.Project != "" {
+			identifier, err := model.GetIdentifierForProject(p.Project)
+			if err != nil {
+				return errors.Wrapf(err, "getting project '%s'", p.Project)
+			}
+			apiPatch.ProjectIdentifier = utility.ToStringPtr(identifier)
+			projectIdentifier = identifier
+		}
+		if args.IncludeCommitQueuePosition && p.IsCommitQueuePatch() {
+			cq, err := commitqueue.FindOneId(p.Project)
+			if err != nil {
+				return errors.Wrap(err, "error getting commit queue position")
+			}
+			apiPatch.CommitQueuePosition = utility.ToIntPtr(-1)
+			if cq != nil {
+				apiPatch.CommitQueuePosition = utility.ToIntPtr(cq.FindItem(p.Id.Hex()))
+			}
+		}
+	}
+	apiPatch.buildBasePatch(p)
+	apiPatch.buildModuleChanges(p, projectIdentifier)
+
+	fmt.Println(len(apiPatch.ModuleCodeChanges))
```

Comment:
do we want to print this?

Reply:
Might be a dumb question but why are we using println if we do?

Reply:
no good question, this is outdated debugging!